include/envoy/http/header_map.h

```diff
@@ -180,11 +172,6 @@ class HeaderEntry {
   virtual void value(const std::string& value) PURE;
 
   /**
-   * Set the header value by moving data into it.
-   */
-  virtual void value(HeaderString&& value) PURE;
```

Comment:
FYI I didn't mean for you to revert all the perf stuff you did. I was just saying to remove `HeaderString& operator=`, rename this to `setMove()`, and then just call `setMove()` wherever you  need it. I'm fine either way though. LMK.

Reply:
Apparently I didn't ctrl-c that push fast enough.

Anyway, I decided this is easier to understand and arguably more correct (the rule is route headers before vhost headers before global headers, so they should come out in the right order even in the face of set-after-append).

It's slightly slower in that it always adds a new HeaderEntry, but I think that's probably swamped by having to iterate over the headers to remove duplicate keys.

Reply:
That said, we could have both if I undid this last commit and then changed the updateByKey search to iterate over the headers in reverse order.

Reply:
OK sounds good. The upside I can see about the update code that you had is it would update in place and not effect ordering. What you have now will move it to the back of the list which is arguably not as good. I don't think it matters that much though.

Reply:
I'm fine either way, just LMK if you are going to change again.